From 5a3a661b9317c46d1ceb9b0d954a56994d88ac07 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Fri, 3 Jul 2020 03:58:28 +0200
Subject: [PATCH 163/163] ARM: dts: stm32: Add PDK2 DTO for DH 638-100 DSI
 mezzanine with RPi 7inch display

Add DT overlay for the PDK2 board for DH 638-100 mezzanine card with
RPi 7" DSI display attached on top.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm/boot/dts/Makefile                    |   1 +
 ...hcom-pdk2-overlay-638-100-x12-rpi7inch.dts | 130 ++++++++++++++++++
 2 files changed, 131 insertions(+)
 create mode 100644 arch/arm/boot/dts/stm32mp157c-dhcom-pdk2-overlay-638-100-x12-rpi7inch.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 25677855fc57..adbe6e665dac 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1006,6 +1006,7 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157c-dhcom-pdk2-overlay-531-100-x21.dtbo \
 	stm32mp157c-dhcom-pdk2-overlay-531-100-x22.dtbo \
 	stm32mp157c-dhcom-pdk2-overlay-560-200-x12.dtbo \
+	stm32mp157c-dhcom-pdk2-overlay-638-100-x12-rpi7inch.dtbo \
 	stm32mp157c-dk2.dtb \
 	stm32mp157c-ed1.dtb \
 	stm32mp157c-ev1.dtb \
diff --git a/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2-overlay-638-100-x12-rpi7inch.dts b/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2-overlay-638-100-x12-rpi7inch.dts
new file mode 100644
index 000000000000..a9de8612277b
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp157c-dhcom-pdk2-overlay-638-100-x12-rpi7inch.dts
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+#include <dt-bindings/gpio/gpio.h>
+
+/dts-v1/;
+/plugin/;
+
+
+/ {
+	fragment-0 {
+		target-path = "/";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		__overlay__ {
+			panel {
+				compatible = "powertip,ph800480t013-idf02";
+				power-supply = <&attiny>;
+				backlight = <&attiny>;
+
+				port {
+					panel_in: endpoint {
+						remote-endpoint = <&bridge_out>;
+					};
+				};
+			};
+		};
+	};
+
+	fragment-1 {
+		target-path = "/soc/i2c@40015000";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			touchscreen@38 {
+				compatible = "edt,edt-ft5406";
+				reg = <0x38>;
+				/*
+				 * Disabled, since the IRQ line is not on
+				 * the FPC cable, so we cannot get touch
+				 * IRQs unless its connected otherwise. In
+				 * that case, add entry like this one and
+				 * enable below.
+				 *
+				 * interrupt-parent = <&gpiog>;
+				 * interrupts = <2 IRQ_TYPE_EDGE_FALLING>;
+				 */
+				status = "disabled";
+			};
+
+			attiny: regulator@45 {
+				compatible = "raspberrypi,7inch-touchscreen-panel-regulator";
+				reg = <0x45>;
+			};
+		};
+	};
+
+	fragment-2 {
+		target-path = "/soc/dsi@5a000000";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			phy-dsi-supply = <&reg18>;
+			status = "okay";
+
+			bridge@0 {
+				compatible = "toshiba,tc358762";
+				reg = <0>;
+				vddc-supply = <&attiny>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				status = "okay";
+
+				port@0 {
+					reg = <0>;
+					bridge_in: endpoint {
+						remote-endpoint = <&dsi_out>;
+					};
+				};
+
+				port@1 {
+					reg = <1>;
+					bridge_out: endpoint {
+						remote-endpoint = <&panel_in>;
+					};
+				};
+			};
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@0 {
+					reg = <0>;
+					dsi_in: endpoint {
+						remote-endpoint = <&ltdc_ep0_out>;
+					};
+				};
+
+				port@1 {
+					reg = <1>;
+					dsi_out: endpoint {
+						remote-endpoint = <&bridge_in>;
+					};
+				};
+			};
+		};
+	};
+
+	fragment-3 {
+		target-path = "/soc/display-controller@5a001000";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		__overlay__ {
+			status = "okay";
+			port {
+				ltdc_ep0_out: endpoint@0 {
+					reg = <0>;
+					remote-endpoint = <&dsi_in>;
+				};
+			};
+		};
+	};
+};
-- 
2.27.0

